conans/test/unittests/util/tools_test.py

```diff
@@ -794,6 +795,30 @@ def vswhere_description_strip_test(self):
             json = vswhere()
             self.assertNotIn("descripton", json)
 
+    @unittest.skipUnless(platform.system() == "Windows", "Requires vswhere")
+    def vswhere_path_test(self):
+        # vswhere not found
+        with tools.environment_append({"ProgramFiles": None, "ProgramFiles(x86)": None, "PATH": ""}):
+            with self.assertRaisesRegex(ConanException, "Cannot locate vswhere"):
+                vswhere()
+        # vswhere in ProgramFiles
+        program_files = get_env("ProgramFiles(x86)") or get_env("ProgramFiles")
+        vswhere_path = None
+        if program_files:
+            expected_path = os.path.join(program_files, "Microsoft Visual Studio", "Installer",
+                                         "vswhere.exe")
+            if os.path.isfile(expected_path):
+                vswhere_path = expected_path
```

Comment:
I don't see this ``vswhere_path`` is used.
I don't understand what this part of the test is testing, that vswhere() can be executed even if it is not in the path? Maybe a comment, "vswhere is in PrgramFiles BUT NOT in the system PATH"

Reply:
Yes, that's it. I use the ``vswhere_path`` variable to locate the real path in the testing machine and then set-up the different scenarios for a real test